Makefile

```diff
@@ -321,8 +321,8 @@ endef
 # don't inadvertently link to /lib/libgcc_s.so.1, which is incompatible with
 # libgfortran, and on Windows we copy them in earlier as well.
 ifeq (,$(findstring $(OS),FreeBSD WINNT))
-julia-base: $(build_libdir)/libgfortran.$(SHLIB_EXT)
-$(build_libdir)/libgfortran.$(SHLIB_EXT): | $(build_libdir)
+julia-base: $(build_libdir)/libgfortran*.$(SHLIB_EXT)*
+$(build_libdir)/libgfortran*.$(SHLIB_EXT)*: | $(build_libdir) julia-deps
```

Comment:
Depending on a pseudo target won’t work properly

Reply:
Hm. Seems to be undocumented, but one SO user at least seems to believe that to be true as well https://stackoverflow.com/a/19687797

Reply:
Which part is undocumented?

Order-only prerequisites [are documented here](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html), showing that this forces building of the `$(build_libdir)/libgfortran*.$(SHLIB_EXT)*` target until after `julia-deps` has run, but not forcing it to rebuild if `$(build_libdir)/libgfortran*.$(SHLIB_EXT)*` is already satisfied.

Wildcard expansion is [documented here](https://www.gnu.org/software/make/manual/html_node/Wildcards.html), with the caveat about non-matching files [being left as un-expanded literal paths here](https://www.gnu.org/software/make/manual/html_node/Wildcard-Pitfall.html#Wildcard-Pitfall).  I find this to pretty much match `bash`'s behavior regarding wildcards.

Reply:
The part about not forcing it to build if julia-deps is not satisfied (since it can never be satisfied)

Reply:
Since `julia-deps` is a rule without a recipe, I figured it would just be replaced with the set of its concrete dependencies.  But that's true, I can't find that documented anywhere.

Reply:
No, that's definitely not what happens (sadly). This is mentioned in the documentation: https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html#Phony-Targets

> A phony target should not be a prerequisite of a real target file; if it is, its recipe will be run every time make goes to update that file. As long as a phony target is never a prerequisite of a real target, the phony target recipe will be executed only when the phony target is a specified goal (see Arguments to Specify the Goals).

Reply:
Right, this is why I specifically said "a rule without a recipe"; because it has no recipe I don't think it counts as a "phony target"; it's just a collection of rules.

This is very easy to test:

```makefile
define touch_rule
$(1):
    @echo "Making $(1)"
    @touch "$(1)"
endef

$(foreach f,a b c,$(eval $(call touch_rule,$(f))))

ghost-rule: a b

ghost-depend: c | ghost-rule
    @echo "ghost-depend"
    @touch "$@"

.PHONY: ghost-rule
```

```
$ make ghost-depend
Making c
Making a
Making b
ghost-depend
$ make ghost-depend
make: `ghost-depend' is up to date.
```

Reply:
If you add a recipe to `ghost-depend`, e.g.:

```makefile
ghost-rule: a b
    echo "ghost-rule"
```

Then running `ghost-depend` will cause `ghost-rule` to be rebuilt every time.

Reply:
It has a recipe (julia-deps it has to). But I guess that’s ok, since we do actually want to force it to run.

Reply:
Ah, you're right.  I had it confused with another rule.  I think you're right though, in this case it doesn't even matter, it's okay to run that again.  Do you have any further comments?